pkg/apis/common/v1/common.go

```diff
@@ -115,7 +115,9 @@ type ObjectSelector struct {
 	ServiceName string `json:"serviceName,omitempty"`
 
 	// SecretName is the name of an existing Kubernetes secret that contains connection information for associating an
-	// Elastic resource not managed by the operator. The referenced secret must contain the following:
+	// Elastic resource not managed by the operator.
+	// SecretName is only supported for monitoring associations. Refer to https://www.elastic.co/docs/deploy-manage/monitor/stack-monitoring/eck-stack-monitoring#k8s_connect_to_an_external_monitoring_elasticsearch_cluster
```

Comment:
Isn't the link to the stack monitoring documentation a bit "restrictive"? We can use `secretName` to setup [Fleet's Elasticsearch output](https://github.com/elastic/cloud-on-k8s/blob/1832f4bbc4e280ffeff5df96fd4ce3bf503899b4/config/e2e/monitoring.yaml#L9-L10) for example. I was wondering if we should not document or validate any limitation on the parent attribute, for example on `FleetServerRef`.

Reply:
That's a good point. I think I overshot the target a bit after the issues our users had with this.